elcid/api.py

```diff
@@ -179,15 +129,84 @@ def is_empty_value(self, observation_value):
         else:
             return observation_value is None
 
+    def get_qs(self, patient):
+        to_ignore = ['UNPROCESSED SAMPLE COMT', 'COMMENT', 'SAMPLE COMMENT']
+        a_year_ago = datetime.date.today() - datetime.timedelta(365)
+        lab_tests = patient.lab_tests.all()
+        lab_tests = lab_tests.exclude(
+            test_name__in=to_ignore
+        )
+        lab_tests = lab_tests.filter(datetime_ordered__gte=a_year_ago)
+        return lab_tests.prefetch_related('observation_set')
+
+    def get_observations_by_type_and_date_str(self, observations):
+        """
+        Returns a dict of observation name -> date -> observation
+        We choose the most recent date for the datetime unless the most
+        recent is 'Pending'
+        """
+        result = defaultdict(dict)
+        observations = sorted(
+            observations, key=lambda x: x.observation_datetime
+        )
+        observations.reverse()
```

Comment:
Doesn't providing your own sorting function only to reverse it on the next line seem odd?
`- ....to_timestamp()` or similar ?

Reply:
So we were doing 

```
        observations = sorted(
            observations, key=lambda x:- x.observation_datetime.toordinal()
        )
```

I think this is the new way is more readable. I think I prefer this way but I'm happy with either.

Reply:
much better idea... use the `reverse=True` argument...